Runtime/Interaction/StaticMeshCollidersManager.cs

```diff
@@ -0,0 +1,223 @@
+                colliderCoords.Add(col, coords);
+                rawCoordsToCollider.Add(new Vector2Int(x, y), col);
+            }
+        }
+
+        public Vector2 GetSurfacePointForCollider(Collider col)
+        {
+            if (!colliderCoords.TryGetValue(col, out Vector2 coordsForCol))
+            {
+                throw new ArgumentException("Unknown {collider.name}");
+            }
+            return coordsForCol;
+        }
+        
+        protected void UpdateColliderPositions()
+        {
+            targetMesh.BakeMesh(tempMesh, true);
+            tempMesh.GetVertices(vertices);
+            tempMesh.GetNormals(normals);
+
+            vertices_native.CopyFrom(vertices.ToArray());
+            normals_native.CopyFrom(normals.ToArray());
+            
+            DeformedCollidersJob job = new DeformedCollidersJob()
+            {
+                Normals = normals_native,
+                Vertices = vertices_native,
+                RemappedIndices = remapped_vertices_data,
+                MaxX = meshXResolution,
+                MaxY = meshYResolution,
+            };
+
+            JobHandle jobHandle = job.Schedule(colliderObjects);
+            jobHandle.Complete();
+        }
+        
+        struct DeformedCollidersJob: IJobParallelForTransform
+        {
+            private Vector3 right, upwards, temppos;
+            public float ScaleFactor, GridSize; 
+            public int MaxX, MaxY;
+
+            [ReadOnly] public NativeArray<Vector3> Vertices;
+            [ReadOnly] public NativeArray<Vector3> Normals;
+            [ReadOnly] public NativeArray<int> RemappedIndices;
+	
+            public void Execute(int i, TransformAccess col)
+            {
+                temppos = Vertices[RemappedIndices[i]];
+                temppos.z += -0.0002f;
+
+                if (i >= (MaxY-1) * MaxX )
+                    upwards = Vertices[RemappedIndices[i - MaxX]] - Vertices[RemappedIndices[i]];
+                else
+                    upwards = Vertices[RemappedIndices[i+ MaxX]] - Vertices[RemappedIndices[i]];
+                
+                col.localPosition = temppos;
+                col.localRotation = Quaternion.LookRotation(Normals[RemappedIndices[i]], upwards);
+            }
+        }
+
+
+        public void RemapVertices()
+        {
+            targetMesh = GetComponent<HPUIStaticContinuousInteractable>().StaticHPUIMesh;
+            tempMesh = new Mesh(); 
+            if(targetMesh==null)
+            {
+                Debug.LogError("Please assign static mesh to the HPUI Static Continuous Interactable Component first!");
+                return;
+            }
+            targetMesh.BakeMesh(tempMesh, true);
+            vertexRemapData = GetRectifiedIndices(vertexRemapData, tempMesh, flipOrderForRecompute);
+        }
+        
+        private static int[] GetRectifiedIndices(int[] remapData, Mesh mesh, bool flipOrder)
```

Comment:
Even though I love pure functions, does this need to be a static method?

Reply:
done